### PR TITLE
IGNITE-8870: Example of usage of encrypted cache.

### PR DESCRIPTION
**Pull request by nizhikov**

### CODE REVIEW

File: examples/src/test/java/org/apache/ignite/testsuites/IgniteExamplesSelfTestSuite.java [hunk 1]
```diff
@@ -101,6 +102,8 @@ public static TestSuite suite() throws Exception {
         // ML Grid.
         suite.addTest(IgniteExamplesMLTestSuite.suite());
 
+        suite.addTest(new TestSuite(EncryptedCacheExampleSelfTest.class));
```
Comment (anton-vinogradov) [inline examples/src/test/java/org/apache/ignite/testsuites/IgniteExamplesSelfTestSuite.java]: use suite.addTestSuite semantic.
see org.apache.ignite.testsuites.IgniteBasicTestSuite for example.

File: examples/src/main/java/org/apache/ignite/examples/encryption/EncryptedCacheExample.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            cache.put(2L, new BankAccount("Middle account", 1_000L));
+            cache.put(3L, new BankAccount("One dollar account", 1L));
+        }
+
+        // After cluster shutdown data persisted on the disk in encrypted form.
+
+        System.out.println(">>> Starting cluster again.");
+        // Starting cluster again.
+        try (Ignite ignite = Ignition.start("examples/config/encryption/example-encrypted-store.xml")) {
+            ignite.active(true);
```
Comment (anton-vinogradov) [inline examples/src/main/java/org/apache/ignite/examples/encryption/EncryptedCacheExample.java]: Deprecated?

File: examples/src/main/java/org/apache/ignite/examples/encryption/EncryptedCacheExample.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            System.out.println(">>> Saved data:");
+
+            // Iterating through existing data.
+            for (Cache.Entry<Long, BankAccount> entry : cursor) {
+                System.out.println(">>> ID = " + entry.getKey() +
+                    ", AccountName = " + entry.getValue().accountName +
+                    ", Balance = " + entry.getValue().balance);
+            }
+
+        }
```
Comment (anton-vinogradov) [inline examples/src/main/java/org/apache/ignite/examples/encryption/EncryptedCacheExample.java]: You have also check that it's impossible to gain data without encryptionSpi.
Seems you have to provide a couple of spring configs.

File: examples/src/main/java/org/apache/ignite/examples/encryption/EncryptedCacheExample.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    public static void main(String[] args) {
+        System.out.println(">>> Starting cluster.");
+
+        // Starting Ignite with EncryptionSpi configured.
+        // Please, note, you should have the same keystore on every server node in cluster with enabled encryption.
+        // You can use encryption feature only for deployment with Ignite persistence enabled.
+        try (Ignite ignite = Ignition.start("examples/config/encryption/example-encrypted-store.xml")) {
+            // Activate the cluster. Required to do if the persistent store is enabled because you might need
+            // to wait while all the nodes, that store a subset of data on disk, join the cluster.
+            ignite.active(true);
```
Comment (anton-vinogradov) [inline examples/src/main/java/org/apache/ignite/examples/encryption/EncryptedCacheExample.java]: Deprecated?

File: examples/src/test/java/org/apache/ignite/testsuites/IgniteExamplesSelfTestSuite.java [hunk 5]
```diff
@@ -101,6 +102,8 @@ public static TestSuite suite() throws Exception {
         // ML Grid.
         suite.addTest(IgniteExamplesMLTestSuite.suite());
```
Comment (anton-vinogradov) [inline examples/src/test/java/org/apache/ignite/testsuites/IgniteExamplesSelfTestSuite.java]: Add comment about test group

File: examples/src/main/java/org/apache/ignite/examples/encryption/EncryptedCacheExample.java (hunk 2 above)
Comment (nizhikov) [reply examples/src/main/java/org/apache/ignite/examples/encryption/EncryptedCacheExample.java]: Fixed

**Pull request closed without merging**